File: arsh_goyal_sde_sheet/arrays/75. Sort Colors/insertion_sort.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        for(int i = 1; i < nums.size(); i++) {
            int cur = nums[i];
            int j = i - 1;
            while (j >= 0 && nums[j] > cur) {
                nums[j + 1] = nums[j];
                j -= 1;
            }
            nums[j+1] = cur;
        }
    }
};

File: arsh_goyal_sde_sheet/arrays/75. Sort Colors/insertion_sort.py
class Solution:
    # insertion sort
    def sortColors(self, nums: List[int]) -> None:
        for i in range(1, len(nums)):
            current = nums[i]
            j = i - 1
            
            while j >= 0 and nums[j] > current:
                nums[j+1] = nums[j]
                j -= 1
            
            nums[j+1] = current
        
        